docs/reference-guide/graph-algorithms.md

```diff
@@ -312,4 +313,79 @@ property equal to `false` and to nodes with a `drinks_USD` property less than `1
 ```cypher
 MATCH path=(n {id: 0})-[*WSHORTEST (r, n | n.total_USD) total_weight (r, n | r.eu_border = false AND n.drinks_USD < 15)]-(m {id: 46})
 RETURN path,total_weight;
+```
+
+## All Shortest Paths
+
+Finding all shortest paths is an expansion of the weighted shortest paths problem. The goal
+of finding the shortest path is obtaining any minimum sum of weights on the path from one
+node to the other. However, there could be multiple similar-weighted paths, and this algorithm
+enables to fetch them all.
+
+Weighted shortest path implementation returns only one-row resulting path from one
+node to the other. Commonly, multiple shortest paths are flowing through different
+routes. Syntax of obtaining all shortest paths is similar to the weighted shortest path
+and boils down to calling `[*ALLSHORTEST (r, n | r.weight)]` where `r` and `n` define 
+the current expansion relationship and node respectively.
+
+### Getting various results
+
+We'll use a similar example as the one above, used for the weighted shortest path.
+To showcase the characteristics of all shortest paths, we'll use a default weight equal to 1 in the next example.
+
+```cypher
+MATCH path=(n {id: 0})-[:CloseTo *ALLSHORTEST (r, n | 1)]-(m {id: 15})
+RETURN path;
+```
+
+Query returns multiple results, all with the 4 hops meaning that there are 
+multiple paths flowing from source node to the destination node.
+
+Nothing prevents the user to run a weighted query based on the weight property on
```

Comment:
I'm not really sure what I get here as a result... maybe rewrite it to:

The following query will return.... ?